Update the Android  OkHttp instrumentation docs and fix some Otel docs (#29392)
DataDog/documentation: content/en/real_user_monitoring/mobile_and_tv_monitoring/android/setup.md

{{% /tab %}}
{{< /tabs >}}

> If you want RUM resources and Spans to be automatically created for your OkHttp requests, you can use the `DatadogInterceptor` as an interceptor.

Reviewer: ```suggestion
To automatically create RUM resources and spans for your OkHttp requests, use the `DatadogInterceptor` as an interceptor.
```

> If you want RUM resources and Spans to be automatically created for your OkHttp requests, you can use the `DatadogInterceptor` as a network interceptor.

Reviewer 2: There is a bit of confusion here by using `network interceptor`, because of the sentence `In case you want to monitor the network redirects or retries, you can use the DatadogInterceptor as a network interceptor.` below.

Is it meant for the `networkInterceptor` usage here or `interceptor`?

Author: ahh mistake, it should be `interceptor` there

This records each request processed by the `OkHttpClient` as a resource, with all the relevant information (URL, method, status code, and error) automatically filled in. Only the network requests that started when a view is active are tracked. To track requests when your application is in the background, [create a view manually][13].

> In case you want to monitor the network redirects or retries, you can use the `DatadogInterceptor` as a network interceptor.

Reviewer: ```suggestion
To monitor the network redirects or retries, you can use the `DatadogInterceptor` as a network interceptor.
```

> In case you want to monitor the network redirects or retries, you can use the `DatadogInterceptor` as a network interceptor.

Reviewer 2: maybe it is worth to link OkHttp documentation on the difference between `interceptor` and `networkInterceptor`?

Author: not sure, I am waiting for suggestions here from doc reviewers.

Reviewer 3: We link to the OkHttp docs early in step 3, right before the tabs that you can see the closing tags for at the very top of this PR. I think that's close enough to this note to be useful, but I'm not against adding the link again if you're concerned about users getting the two confused.

Reviewer: Yes, linking to the OkHttp docs is a good idea. I would put the link on the phrase "network interceptor."

> If you want RUM resources and Spans to be automatically created for your OkHttp requests, you can use the `DatadogInterceptor` as an interceptor.
> 
> This records each request processed by the `OkHttpClient` as a resource, with all the relevant information (URL, method, status code, and error) automatically filled in. Only the network requests that started when a view is active are tracked. To track requests when your application is in the background, [create a view manually][13].
> 
> In case you want to monitor the network redirects or retries, you can use the `DatadogInterceptor` as a network interceptor.
> 
> {{< tabs >}}
> {{% tab "Kotlin" %}}
> ```kotlin
> val okHttpClient = OkHttpClient.Builder()
>     .addNetworkInterceptor(DatadogInterceptor.Builder(tracedHostsWithHeaderType).build())
>     .build()
> ```
> {{% /tab %}}
> {{% tab "Java" %}}
> ```java
> OkHttpClient okHttpClient = new OkHttpClient.Builder()
>     .addNetworkInterceptor(new DatadogInterceptor.Builder(tracedHostsWithHeaderType).build())
>     .build();
> ```
> {{% /tab %}}
> {{< /tabs >}}

Reviewer 3: The two notes about using `DatadogInterceptor` seem similar to each other; I'm wondering if someone reading quickly might skip "network" and get confused.

I'm not sure I totally understand the intended placement of the paragraphs as they are now; I suspect that more should be embedded under step 3 than currently is, and that we can use some further embedding to make the relationships between these pieces more clear. I was wondering if this would make sense to you:

```suggestion
- If you want RUM resources and Spans to be automatically created for your OkHttp requests, you can use the `DatadogInterceptor` as an interceptor.
  - This records each request processed by the `OkHttpClient` as a resource, with all the relevant information (URL, method, status code, and error) automatically filled in. Only the network requests that started when a view is active are tracked. To track requests when your application is in the background, [create a view manually][13].
- In case you want to monitor the network redirects or retries, you can use the `DatadogInterceptor` as a network interceptor:

  {{< tabs >}}
  {{% tab "Kotlin" %}}
  \```kotlin
  val okHttpClient = OkHttpClient.Builder()
      .addNetworkInterceptor(DatadogInterceptor.Builder(tracedHostsWithHeaderType).build())
      .build()
  \```
  {{% /tab %}}
  {{% tab "Java" %}}
  \```java
  OkHttpClient okHttpClient = new OkHttpClient.Builder()
      .addNetworkInterceptor(new DatadogInterceptor.Builder(tracedHostsWithHeaderType).build())
      .build();
  \```
  {{% /tab %}}
  {{< /tabs >}}
```

**Note** that in order to make that suggestion render properly in GitHub, I had to escape the code samples, so if you accept that change, you have to un-escape them. I don't know how to make that less buggy 😬

This is what that would look like:
<img width="770" alt="image" src="https://github.com/user-attachments/assets/aee33cde-226e-4e4b-a891-5c57648bfd0b" />

Author: This looks great thanks !!

Author: @janine-c can you try to add again the suggestion ? It was collapsing with another suggestion that I already commit and I am not sure exactly what and where should I change now to apply a new format :). Tks !!

> **Note**: In case you only want to have Spans without RUM resources, you can use the `TracingInterceptor` instead of `DatadogInterceptor` the same way we explained above.

Reviewer: ```suggestion
**Note**: To use spans but not RUM resources, you can use the `TracingInterceptor` instead of `DatadogInterceptor` as described above.
```

> **Note** In case you only want to have Spans without RUM resources, you can use the `TracingInterceptor` instead of `DatadogInterceptor` the same way we explained above.

Reviewer 2: ```suggestion
**Note**: In case you only want to have Spans without RUM resources, you can use the `TracingInterceptor` instead of `DatadogInterceptor` the same way we explained above.
```

> **Note**: In case you only want to have Spans without RUM resources, you can use the `TracingInterceptor` instead of `DatadogInterceptor` the same way we explained above.
> 
> **Note**: If you also use multiple Interceptors, add `DatadogInterceptor` first.

Reviewer 3: ```suggestion
**Notes**:
- If you only want to have Spans without RUM resources, you can use the `TracingInterceptor` in the place of the `DatadogInterceptor`.
- If you also use multiple Interceptors, add `DatadogInterceptor` first.
```
